draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1991,36 +2012,201 @@ conditions.
+These suites share the following parameters:
+
+- E: y^2 = x^3 + A * x + B, where
+  - A = -3
+  - B = 0x51953eb9618e1c9a1f929a21a0b68540eea2da725b99b315f3b8b489918ef109e156193951ec7e937b1652c0bd3bb1bf073573df883d2c34f1ef451fd46b503f00
+- p: 2^521 - 1
+- m: 1
+- H: SHA-512
+- W: 2
+- f: Simplified SWU method, {{simple-swu}}
+- Z: -2
+- h\_eff: 1
+
+## Suites for curve25519 and edwards25519 {#suites-25519}
+
+This section defines ciphersuites for curve25519 and edwards25519 {{RFC7748}}.
+
+The suites curve25519-SHA256-ELL2-RO and curve25519-SHA256-ELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: y^2 = x^3 + 486662 * x^2 + x
+- f: Elligator 2 method, {{elligator2}}
+
+The suites edwards25519-SHA256-EDELL2-RO and edwards25519-SHA256-EDELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: -x^2 + y^2 = 1 + d * x^2 * y^2, where
+  - d = 0x52036cee2b6ffe738cc740797779e89800700a4d4141d8ab75eb4dca135978a3
+- f: Twisted Edwards Elligator 2 method, {{ell2edwards}}
+- M: curve25519 defined in {{RFC7748}}, Section 4.1
+- rational\_map: the birational map defined in {{RFC7748}}, Section 4.1
+
+The common parameters for all of the above suites are:
+
+- p: 2^255 - 19
+- m: 1
+- H: SHA-256
+- W: 2
+- Z: 2
+- h\_eff: 8
+
+## Suites for curve448 and edwards448 {#suites-448}
+
+This section defines ciphersuites for curve448 and edwards448 {{RFC7748}}.
+
+The suites curve448-SHA512-ELL2-RO and curve448-SHA512-ELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: y^2 = x^3 + 156326 * x^2 + x
+- f: Elligator 2 method, {{elligator2}}
+
+The suites edwards448-SHA512-EDELL2-RO and edwards448-SHA512-EDELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: x^2 + y^2 = 1 - 39081 * x^2 * y^2
+- f: Twisted Edwards Elligator 2 method, {{ell2edwards}}
+- M: curve448, defined in {{RFC7748}}, Section 4.2
+- rational\_map: the 4-isogeny map defined in {{RFC7748}}, Section 4.2
+
+The common parameters for all of the above suites are:
+
+- p: 2^448 - 2^224 - 1
+- m: 1
+- H: SHA-512
+- W: 2
+- Z: -1
+- h\_eff: 4
+
+## Suites for SECP256K1 {#suites-secp256k1}
+
+The suites SECP256K1-SHA256-SVDW-RO and SECP256K1-SHA256-SVDW-NU
+are defined for the SECP256K1 elliptic curve {{SEC2}}.
+These suites share the following parameters:
+
+- E: y^2 = x^3 + 7
+- p: 2^256 - 2^32 - 2^9 - 2^8 - 2^7 - 2^6 - 2^4 - 1
+- m: 1
+- H: SHA-256
+- W: 2
+- f: Shallue-van de Woestijne method, {{swpairing}}
+- Z: 1
+- h\_eff: 1
+
+## Suites for BLS12-381 {#suites-bls12381}
+
+This section defines ciphersuites for groups G1 and G2 of
+the BLS12-381 elliptic curve {{draft-yonezawa-pfc-01}}.
+
+The suites BLS12\_381\_G1-SHA256-SSWU-RO and BLS12\_381\_G1-SHA256-SSWU-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: y^2 = x^3 + 4
+- m: 1
+- Z: -1
+- E': the 11-isogenous curve from {{WB19}}; see {{appx-bls12381-g1}}
```

Comment:
the curve equation of `E'` must be stated explicitly.

Reply:
Right now it's in the appendix. Do you prefer to pull it inline here, and only have the isogeny maps in the appendix?

Reply:
I prefer to have suite descriptions as explicit as possible.
For the case of the 11-isogeny, you can list only the point generator of the kernel, and cite to Velú formulas for calculating the isogeny, which I assume is the method followed by the Sage script you provide in the paper.

Reply:
It seems like that could work, but I have a couple concerns:

- the target audience is implementors, who may not be familiar with Velu's formulas. My guess is that most would prefer being given the rational map explicitly rather than a more compact description that takes a lot more work to turn into code, and thus is more likely to cause errors.

- specifying the kernel of the isogeny only specifies the rational map up to sign: negating the sign of the y-coordinate map must give an isogeny with the same kernel (because negating the point at infinity yields the point at infinity, and negating any other point does not).

I suppose we could specify both the kernel and the rational map, but it's not obvious that the extra information is particularly useful: if someone *really* wants the kernel, they can easily compute it from E, E', and the degree of the isogeny, or they can just go look in the paper...

Reply:
> I prefer to have suite descriptions as explicit as possible.

E' is given in an appendix of *this document* (see e7b4fb9), which seems to me to be no less explicit than putting it in this section. But I think it's totally reasonable to put the description of E' here, if you think that's better.

Reply:
I pushed a commit that pulls E' definitions inline and only refers to the appx for the iso_map definitions.

Reply:
it looks good